src/livekit/MediaDevicesContext.tsx

```diff
@@ -84,15 +85,19 @@ function useMediaDevice(
     [kind, requestPermissions],
   );
   const available = useObservableState(deviceObserver, []);
-  const [selectedId, select] = useState(fallbackDevice);
+  const [selectedId, select] = useSetting(setting);
 
   return useMemo(() => {
-    let devId;
-    if (available) {
-      devId = available.some((d) => d.deviceId === selectedId)
-        ? selectedId
-        : available.some((d) => d.deviceId === fallbackDevice)
-          ? fallbackDevice
+    let devId: string | undefined = undefined;
+    if (!alwaysDefault && available) {
+      // If the selected device is available, use it. Or if every available
+      // device ID is falsy, the browser is probably just being paranoid about
+      // fingerprinting and we should still try using the selected device.
+      // Otherwise, fall back to the first available device.
+      devId =
+        available.some((d) => d.deviceId === selectedId) ||
+        available.every((d) => d.deviceId === "")
+          ? selectedId
           : available.at(0)?.deviceId;
     }
```

Comment:
Where is the fallback happening?
If all devices are `""` we would still end up using selectedId. What if that is not available? If the browser is paranoid but also the mic has been disconnected?
Where do we catch that?